cyclegan_arch/mnist_to_svhn/solver.py

```diff
@@ -231,6 +235,160 @@ def set_expectation_and_std(self):
 
     def as_np(self, data):
         return data.cpu().data.numpy()
+    
```

Comment:
The new test function.
Copied mostly from the train function. 
No grad updates and uses test loader.
Calculates all the same losses.